### PR TITLE
adding RTD conf file V2 as per request

### PR DESCRIPTION
**Pull request by esc**
As title

### CODE REVIEW

**Comment by esc** [comment]
This is failing now, even though I added a `.readthdocs.yaml` file as appropriate.

Screen Shot 2023-08-15 at 14 59 10

**Comment by esc** [comment]
I cherry-picked 🍒 6e50882fa22f4bc3b3bd4e1f588937773bb91d40 from #983 in order to try to fix building on RTD.

File: .readthedocs.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# .readthedocs.yaml
+# Read the Docs configuration file
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+version: 2
+
+build:
+  os: ubuntu-22.04
```
Comment (jezdez) [inline .readthedocs.yml:8]:
```suggestion
  os: "ubuntu-22.04"
```

Comment (esc) [reply .readthedocs.yml]: I don't think that is the problem. The string `ubuntu-22.04` is also listed w/o quotes here:

https://docs.readthedocs.io/en/stable/config-file/v2.html

Comment (esc) [reply .readthedocs.yml]: Also, https://www.yamllint.com/ claims the `.readthedocs.yaml` files is indeed valid YAML.

Comment (jezdez) [reply .readthedocs.yml]: Just noting that it DOES work for https://github.com/conda/conda-build/pull/5006/files#diff-cde814ef2f549dc093f5b8fc533b7e8f47e7b32a8081e0760e57d5c25a1139d9R4

Comment (jezdez) [reply .readthedocs.yml]: (could also just be an unrelated state issue since you're adding the file completely new)

**Comment by esc** [comment]
After removal of the old `readthedocs.yaml` file (without the `.` prefix) I now get:

```
Processing /home/docs/checkouts/readthedocs.org/user_builds/llvmlite/checkouts/983
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      setup.py disabled on readthedocs: called with ['/home/docs/checkouts/readthedocs.org/user_builds/llvmlite/checkouts/983/setup.py', 'egg_info', '--egg-base', '/tmp/pip-pip-egg-info-bp1l4alb']
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

**Comment by esc** [comment]
Notes for reviewers:

* check correct theme on local and RTD builds
* check string `last updated..` is injected correctly in both environments
* overall sanity checks and look for more opportunities to remove legacy

File: docs/environment.yml [hunk 2]
```diff
@@ -1,8 +0,0 @@
-# Conda environment configuration for readthedocs
-
-name: rtdenv
-dependencies:
-  - python=3.9
-  - sphinx
-  # https://stackoverflow.com/questions/67542699/readthedocs-sphinx-not-rendering-bullet-list-from-rst-fileA
-  - docutils==0.16
```
Comment (sklam) [inline docs/environment.yml:8]: checked bullet list renders correctly.

**Review by sklam — APPROVED** [review]
The doc rendering checks out.

**Pull request merged**